sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -104,6 +119,14 @@ class InMemoryCatalog extends ExternalCatalog {
         }
       }
       // Remove the database.
+      val dbDefinition = catalog(db).db
+      try {
+        fs.delete(new Path(dbDefinition.locationUri), true)
```

Comment:
So far, in `HiveExternalCatalog`, I saw this issue. Hive metastore has the same behavior. 

Regarding `Purge`, DROP DATABASE does not have such an option.